### ISSUE TITLE
Cannot access any chats in purple-matrix

### ISSUE THREAD

**Issue by bodqhrohro** [SpectrumIM/spectrum2#420]
The registration and login seem to work successfully, according to warnings about certain chats appearing in Spectrum2 log, but no any buddies or rooms appear in the roster, and I don't see them in Service Discovery as well.

I would fix it maybe, but I have no idea what is wrong exactly and what should prpls implement correctly so Spectrum2 can obtain the chats list. Does maybe purple-matrix rely on an ability to spawn a chat tab before populating the buddy list, which works well in Pidgin/Finch, but not in Spectrum2?

**Comment by vitalyster** [comment]
Chats should be sent using roomlist api, you can check other prpls like purple-skypeweb for example

**Comment by bodqhrohro** [comment]
Thanks, I'll look into it, seems like there's not much code to add.

And also, is it possible to adjust the loglevel? This prpl is extremely noisy and has spawned 2GB of INFO messages in syslog (status changes and so) just during one night.

**Comment by bodqhrohro** [comment]
I [implemented](https://github.com/bodqhrohro/purple-matrix/commit/e2de17bdd1c9666f58a1acc31b082dd2661dd653) populating the room list from open conversations. It works in Pidgin:
![2021-12-14-045147_872x396_scrot](https://user-images.githubusercontent.com/2134486/145924737-c2cd45fe-26f5-4753-805f-9e16e3b146b2.png)
, but rooms still don't appear in service discovery:
![2021-12-14-045607_1366x768_scrot](https://user-images.githubusercontent.com/2134486/145925038-fc53b760-9736-4a8d-851d-54f1ce37def7.png)

Any other suggestions please?

Actually, this prpl is already pretty annoying due to the fact it opens a conversation for every existing room on every connection, and relies on it, unlike other prpls do: this is why I avoid using it in Pidgin. And it doesn't even seem to cache the rooms anywhere out of structures it provides for libpurple. If Spectrum2 doesn't actually emulate conversations spawning, there still should be just an empty list. Should I maybe explicitly trigger a sync and retrieve the rooms from its results, like it's done for XMPP and Skype?

*About the logging:* I tried to circumvent it by replacing `log4j.rootLogger=debug, R` by `log4j.rootLogger=WARN, R` in all applicable `.cfg` files. Still not sure if it works correctly, as there is still a plenty of logs.

**Comment by vitalyster** [comment]
You are seeing output from libpurple, you should configure libpurple logging, not spectrum. Spectrum just starts libpurple at debug level https://github.com/SpectrumIM/spectrum2/blob/master/backends/libpurple/main.cpp#L2410

**Comment by vitalyster** [comment]
The general issue with Matrix is: matrix IDs are case-sensitive while email/jid are not. So it is not possible to map matrix id to jid directly. Maybe need to use case_insensitive_hash(matrix_username)@matrix.server to prevent case issues and automatically convert everything, but this will create problems to users who wants to write to matrix ID directly

**Comment by bodqhrohro** [comment]
>Spectrum just starts libpurple at debug level

Yeah, and ignores the `level` argument provided by just passing everything to Log4j at `INFO` level.

Anyway, this is weird itself, as with the WARN level in the config all of that had to be ignored completely, which is not the case. And according to these `.cfg` files, logging should go to the `/var/log/spectrum2` directory rather than to syslog, but this directory does not even exist, so these configs are barely taken into account at all. Where should I configure it then? I'm also not aware if there is a libpurple-level debugging level option in `prefs.xml`.

I thought of resorting to just patching out all of debug output off the prpl, but it's not really a solution.

> matrix IDs are case-sensitive while email/jid are not

AFAIR, Hangouts has the same issue, and still it works somehow, but only via the Service Discovery, not by directly contacting any JIDs.

Anyway, if this has to be addressed specifically, uppercase characters may just be escaped, like `!aBaBaGaLaMagA => !a[b]a[b]a[g]a[l]a[m]ag[a]`. Square brackets are allowed in JIDs, but not in MXIDs. No need in inobvious hashing.
